Few typing issues

<!--
If you need help with discord.js installation or usage, please go to the discord.js Discord server instead:
  https://discord.gg/bRCvFy9
This issue tracker is only for bug reports and enhancement suggestions.
You won't receive any basic help here.
-->

**Please describe the problem you are having in as much detail as possible:**
tsc reports errors with the typings file

**Include a reproducible code sample here, if possible:**
```js
import { Client } from 'discord.js';

new Client()
	.on('ready', () => console.log('Ready'))
	.login(process.env.TOKEN);
```

**Further details:**
- discord.js version: d2da771e0f28773587766ede94cfdae222deefa0
- Node.js version: 10.10.0
- Operating system: Windows 10 Pro x64 1803
- Priority this issue should have – please be realistic and elaborate if possible: Minor

<!--
If either of these applies to you, please check the respective checkbox: [ ] becomes [x].
You don't have to modify the text to suit your particular situation – if you want to
elaborate, please do so in the description.
While it's not a requirement to test your issue on the master branch, it would make fixing
the problem a lot easier for us, so please do so if possible.
-->
- [ ] I found this issue while running code on a __user account__
- [x] I have also tested the issue on latest master, commit hash: d2da771e0f28773587766ede94cfdae222deefa0

Extension of #2835 but not locked, with more input
Here's a video I made to help show it, though it's just showing the same errors as below:
https://youtu.be/GvgDsJeEB8I

tsc output:
```
node_modules/discord.js/typings/index.d.ts:38:15 - error TS2417: Class static side 'typeof ActivityFlags' incorrectly extends base class static side 'typeof BitField'.
  Types of property 'resolve' are incompatible.
    Type '(permission: BitFieldResolvable<ActivityFlagsString>) => number' is not assignable to type '(bit?: string | number | Readonly<BitField<string>> | RecursiveArray<string | number | Readonly<BitField<string>>> | undefined) => number'.
      Types of parameters 'permission' and 'bit' are incompatible.
        Type 'string | number | Readonly<BitField<string>> | RecursiveArray<string | number | Readonly<BitField<string>>> | undefined' is not assignable to type 'BitFieldResolvable<ActivityFlagsString>'.
          Type 'undefined' is not assignable to type 'BitFieldResolvable<ActivityFlagsString>'.

38  export class ActivityFlags extends BitField<ActivityFlagsString> {
                 ~~~~~~~~~~~~~

node_modules/discord.js/typings/index.d.ts:789:15 - error TS2417: Class static side 'typeof Permissions' incorrectly extends base class static side 'typeof BitField'.
  Types of property 'resolve' are incompatible.
    Type '(permission?: number | "CREATE_INSTANT_INVITE" | "KICK_MEMBERS" | "BAN_MEMBERS" | "ADMINISTRATOR" | "MANAGE_CHANNELS" | "MANAGE_GUILD" | "ADD_REACTIONS" | "VIEW_AUDIT_LOG" | "PRIORITY_SPEAKER" | ... 22 more ... | undefined) =>
number' is not assignable to type '(bit?: string | number | Readonly<BitField<string>> | RecursiveArray<string | number | Readonly<BitField<string>>> | undefined) => number'.
      Types of parameters 'permission' and 'bit' are incompatible.
        Type 'string | number | Readonly<BitField<string>> | RecursiveArray<string | number | Readonly<BitField<string>>> | undefined' is not assignable to type 'number | "CREATE_INSTANT_INVITE" | "KICK_MEMBERS" | "BAN_MEMBERS" | "ADMINISTRATOR" | "MANAGE_CHANNELS" | "MANAGE_GUILD" | "ADD_REACTIONS" | "VIEW_AUDIT_LOG" | "PRIORITY_SPEAKER" | ... 22 more ... | undefined'.
          Type 'string' is not assignable to type 'number | "CREATE_INSTANT_INVITE" | "KICK_MEMBERS" | "BAN_MEMBERS" | "ADMINISTRATOR" | "MANAGE_CHANNELS" | "MANAGE_GUILD" | "ADD_REACTIONS" | "VIEW_AUDIT_LOG" | "PRIORITY_SPEAKER" | ... 22 more ... | undefined'.

789  export class Permissions extends BitField<PermissionString> {
                  ~~~~~~~~~~~

node_modules/discord.js/typings/index.d.ts:1018:15 - error TS2417: Class static side 'typeof Speaking' incorrectly extends base class static side 'typeof BitField'.
  Types of property 'resolve' are incompatible.
    Type '(permission: BitFieldResolvable<SpeakingString>) => number' is not assignable to type '(bit?: string | number | Readonly<BitField<string>> | RecursiveArray<string | number | Readonly<BitField<string>>> | undefined) => number'.
      Types of parameters 'permission' and 'bit' are incompatible.
        Type 'string | number | Readonly<BitField<string>> | RecursiveArray<string | number | Readonly<BitField<string>>> | undefined' is not assignable to type 'BitFieldResolvable<SpeakingString>'.
          Type 'undefined' is not assignable to type 'BitFieldResolvable<SpeakingString>'.

1018  export class Speaking extends BitField<SpeakingString> {
```
tsconfig.json:
```
{
	"compileOnSave": true,
	"compilerOptions": {
		"allowJs": true,
		"alwaysStrict": true,
		"experimentalDecorators": true,
		"module": "commonjs",
		"moduleResolution": "node",
		"noImplicitAny": true,
		"noUnusedParameters": true,
		"outDir": "build",
		"pretty": true,
		"sourceMap": true,
		"strict": true,
		"target": "esnext"
	},
	"exclude": [
		"node_modules"
	],
	"include": [
		"src",
		"typings"
	]
}
```
Related package.json sections:
```
	"main": "build/index.js",
	"scripts": {
		"prestart": "tsc",
		"start": "node .",
		"pretest": "tsc",
		"test": "node --inspect .",
		"debug": "nodemon -e ts --exec npm test"
	},
	"dependencies": {
		"@types/node": "^10.7.1",
		"discord.js": "github:discordjs/discord.js",
		"typescript": "^3.0.1"
	},
	"devDependencies": {
		"nodemon": "^1.18.3",
		"tslint": "^5.11.0"
	},
```
Let me know if there's anything else I can provide.

--- a/typings/index.d.ts
+++ b/typings/index.d.ts
@@ -37,7 +37,7 @@ declare module 'discord.js' {
 
 	export class ActivityFlags extends BitField<ActivityFlagsString> {
 		public static FLAGS: Record<ActivityFlagsString, number>;
-		public static resolve(permission: BitFieldResolvable<ActivityFlagsString>): number;
+		public static resolve(bit?: BitFieldResolvable<ActivityFlagsString>): number;
 	}
 
 	export class APIMessage {
@@ -109,7 +109,7 @@ declare module 'discord.js' {
 		public valueOf(): number;
 		public [Symbol.iterator](): Iterator<S>;
 		public static FLAGS: object;
-		public static resolve(bit?: BitFieldResolvable<string>): number;
+		public static resolve(bit?: BitFieldResolvable<any>): number;
 	}
 
 	export class CategoryChannel extends GuildChannel {
@@ -1017,7 +1017,7 @@ declare module 'discord.js' {
 
 	export class Speaking extends BitField<SpeakingString> {
 		public static FLAGS: Record<SpeakingString, number>;
-		public static resolve(permission: BitFieldResolvable<SpeakingString>): number;
+		public static resolve(bit?: BitFieldResolvable<SpeakingString>): number;
 	}
 
 	export class Structures {

